scripts/bash_lib.sh

```diff
@@ -528,4 +537,133 @@ check_upgraded_versions() {
+    # some adjustments to them to avoid false positives, but we want original files
+    # to be stored by buildbot
+    for f in ldd-main ldd-columnstore reqs-main reqs-columnstore plugins capabilities; do
+      if [ -e "./${f}.old" ] ; then
+        cp "./${f}.old" "./${f}.old.cmp"
+        cp "./${f}.new" "./${f}.new.cmp"
+      fi
+    done
+
+    # Permanent adjustments
+    sed -i '/libstdc++.so.6(GLIBCXX/d;/libstdc++.so.6(CXXABI/d' ./reqs-*.cmp
+    # Don't compare subversions of libsystemd.so (e.g. libsystemd.so.0(LIBSYSTEMD_227)(64bit))
+    sed -i '/libsystemd.so.[0-9]*(LIBSYSTEMD_/d' ./reqs-*.cmp
+    # Ignore shells, the number of which always changes
+    sed -i '/^\/bin\/sh$/d' ./reqs-*.cmp
+
+    # Here is the place for temporary adjustments,
+    # when legitimate changes in dependencies happen between minor versions.
+    # The adjustments should be done to .cmp files, and removed after the release
+    #
+    # End of temporary adjustments
+
+    bb_log_info "Comparing old and new ldd output for installed binaries"
+    # We are not currently comparing it for Columnstore, too much trouble for nothing
+    if ! diff -U1000 ./ldd-main.old.cmp ./ldd-main.new.cmp | (grep -E '^[-+]|^ =' || true) ; then
+      bb_log_error "Found changes in ldd output on installed binaries"
+      res=1
+    else
+      bb_log_info "OK"
+    fi
+
+    bb_log_info "Comparing old and new package requirements"
+    # We are not currently comparing it for Columnstore, but we may need to in the future
+    if ! diff -U150 ./reqs-main.old.cmp ./reqs-main.new.cmp | (grep -E '^ [^ ]|^[-+]' || true) ; then
+      bb_log_error "Found changes in package requirements"
+      res=1
+    else
+      bb_log_info "OK"
+    fi
+
+    bb_log_info "Comparing old and new server capabilities ('%have%' variables)"
+    if ! diff -u ./capabilities.old.cmp ./capabilities.new.cmp ; then
+      bb_log_error "ERROR: found changes in server capabilities"
+      res=1
+    else
+      bb_log_info "OK"
+    fi
+
+    echo "Comparing old and new plugins"
+    # Columnstore version changes often, we'll ignore it
+    grep -i columnstore ./plugins.old.cmp > /tmp/columnstore.old
+    grep -i columnstore ./plugins.new.cmp > /tmp/columnstore.new
+    if ! diff -u /tmp/columnstore.old /tmp/columnstore.new ; then
+      bb_log_warn "Columnstore version changed. Downgraded to a warning as it is usually intentional"
+      sed -i '/COLUMNSTORE/d;/Columnstore/d' ./plugins.*.cmp
+    fi
+
+    if ! diff -u ./plugins.old.cmp ./plugins.new.cmp ; then
+      bb_log_error "Found changes in available plugins"
+      res=1
+    else
+      bb_log_info "OK"
+    fi
+  fi
+  exit $res
+}
+
+# Collect package requirements and ldd for all binaries included into packages.
+# Expects "old" and "new" as an argument, and
+# $package_list to be set by the time of execution
+collect_dependencies() {
+  old_or_new=$1
+  case "$package_list" in
```

Comment:
This does not work since the extension is not in the `package_list`. You could probably use a switch based on os detection (we usually source `/etc/os-release`, see above). Or what has proven to be quite reliable to me is the following (to be adapted):

```bash
# determine pkg manager
if command -v dnf >/dev/null; then
  pkg_cmd="dnf"
elif command -v yum >/dev/null; then
  pkg_cmd="yum"
elif command -v zypper >/dev/null; then
  pkg_cmd="zypper"
else
  pkg_cmd="apt-get"
fi

```

Reply:
Right. Given that the subroutine is called explicitly from either rpm-upgrade.sh or deb-upgrade.sh and is not used otherwise, and the package type is obviously specific for each of the scripts, I've just made them pass over the package type explicitly as an argument.